### ISSUE TITLE
string & Type intersection produce error in switch statement

### ISSUE THREAD

**Issue by mdaveynis** [Microsoft/TypeScript#10892]
**TypeScript Version:**  2.0.2

**Code**

``` ts
class MyBrand
{
    private _a;
}

function test(strInput: string & MyBrand, numInput: number & MyBrand) {
    switch(strInput)
    {
        case "a": // error: Type '"a"' is not comparable to type 'string & MyBrand'
        return 1;
    }

    switch(numInput)
    {
        case 1: // error: Type '1' is not comparable to type 'number & MyBrand'
        return 1;
    }

    return 0;
}
```

**Expected behavior:**
No error reported in case statements.

**Actual behavior:** 
See above error report inline.

**Comment by nippur72** [comment]
Why do you expect no error? Member `_a` is private but it must be present. In other words `MyBrand` is not an alias for `{}` just because `_a` is private.

**Comment by iskiselev** [comment]
Before 2.0.2 it was a way to create branded strings or numbers.
So, it was possible to create method, that will accept only some strings. For example, I was able to mark introduce type ProductId = string & ProductBrand. After it, objects that was returned by method with ProductId return type was accepted by any method that accept string, but it was possible to create method that accept only ProductId string.
With 2.0.2 this functionality is partially broken.

**Comment by falsandtru** [comment]
also

```ts
type Id = {id:never} & number;
<Id>1 > <number>0;
<Id>1 > 0;
```

```
$ node built/local/tsc.js -t es5 -m commonjs --lib es6 --strictNullChecks index.ts && node index.js
index.ts(3,1): error TS2365: Operator '>' cannot be applied to types 'Id' and '0'.
```

**Comment by sandersn** [comment]
Note that this works with `if`:

```ts
    if (strInput === 'a') {
        return 1;
    }
    if (numInput === 1) {
        return 1;
    }
```

Does not give any errors. `switch` should work the same as `if` with `===`.
